src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -0,0 +1,86 @@
+#include "software/sensor_fusion/sensor_fusion.h"
+
+#include "software/constants.h"
+#include "software/parameter/dynamic_parameters.h"
+
+SensorFusion::SensorFusion()
+    : field_state(0, 0, 0, 0, 0, 0, 0, Timestamp::fromSeconds(0)),
+      ball_state(Point(), Vector(), Timestamp::fromSeconds(0)),
+      friendly_team_state(Duration::fromMilliseconds(
+          Util::Constants::ROBOT_DEBOUNCE_DURATION_MILLISECONDS)),
+      enemy_team_state(Duration::fromMilliseconds(
+          Util::Constants::ROBOT_DEBOUNCE_DURATION_MILLISECONDS)),
+      ball_filter(BallFilter::DEFAULT_MIN_BUFFER_SIZE,
+                  BallFilter::DEFAULT_MAX_BUFFER_SIZE),
+      friendly_team_filter(),
+      enemy_team_filter()
+{
+}
+
+void SensorFusion::onValueReceived(RefboxData refbox_data)
+{
+    updateWorld(refbox_data);
+    Subject<World>::sendValueToObservers(world);
+}
+
+void SensorFusion::onValueReceived(RobotStatus robot_status)
+{
+    updateWorld(robot_status);
+    Subject<World>::sendValueToObservers(world);
+}
+
+void SensorFusion::onValueReceived(VisionDetection vision_detection)
+{
+    updateWorld(vision_detection);
+    Subject<World>::sendValueToObservers(world);
+}
+
+void SensorFusion::updateWorld(RefboxData refbox_data)
+{
+    world.updateRefboxData(refbox_data);
+}
+
+void SensorFusion::updateWorld(RobotStatus robot_status) {}
+
+void SensorFusion::updateWorld(VisionDetection vision_detection)
+{
+    // Parse Detections
```

Comment:
could we have a `parseDetections` function?

Reply:
had to refactor in a different way